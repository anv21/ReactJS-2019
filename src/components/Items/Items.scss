.items_section {
    background-color: #fff;

    .items_wrapper {
        display: flex;
        flex-wrap: wrap;
        width: 900px;
        margin: 0 auto;
    }
    
    .items_empty_result {
        font-size: 50px;
        height: 400px;
        padding-top: 100px;
        width: 100%;
        text-align: center;
    }
}

