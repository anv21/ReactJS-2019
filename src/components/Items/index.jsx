import React from 'react';
import PropTypes from 'prop-types';
import Item from '../Item/index.jsx';
import {connect} from 'react-redux';

const Items = (props) => (
    <div className="items_section">
        <div className="items_wrapper">
            {props.items && props.items.length ? (
                props.items.map(item => (
                    <Item
                        key={item.id}
                        item={item}
                        getItem={props.getItem}
                        getItemsByGenre={props.getItemsByGenre}
                    />
                ))
                ) : (
            <div className="items_empty_result">No films found</div>
            )}
        </div>
    </div>
)

Items.propTypes = {
    items: PropTypes.array
};

Items.defaultProps = {
    items: []
};

const mapStateToProps = (state) => {
    const {items} = state.appReducer;
    return {items};
};

 export default connect(mapStateToProps)(Items);