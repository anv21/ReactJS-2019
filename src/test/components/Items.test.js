import React from 'react';
import { shallow } from 'enzyme';

import Items from '../../components/Items/index.jsx';

describe('Items', () => {
    const items = shallow(<Items />);

    it('should be like snapshot', () => {
        expect(items).toMatchSnapshot();
    });
});
