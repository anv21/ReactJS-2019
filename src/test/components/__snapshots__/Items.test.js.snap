// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Items should be like snapshot 1`] = `ShallowWrapper {}`;
